### PR TITLE
parser: support disable window function parser

### PR DESCRIPTION
**Pull request by lamxTyler**
Change the `EnableWindowFunc` to `SetWindowFunc`, so we can disable it dynamically.

### CODE REVIEW

File: lexer.go [hunk 1]
```diff
@@ -195,9 +195,9 @@ func (s *Scanner) GetSQLMode() mysql.SQLMode {
 	return s.sqlMode
 }
 
-// EnableWindowFunc enables the scanner to recognize the keywords of window function.
-func (s *Scanner) EnableWindowFunc() {
-	s.supportWindowFunc = true
+// SetWindowFunc controls whether the scanner recognize the keywords of window function.
+func (s *Scanner) SetWindowFunc(val bool) {
```
Comment (zz-jason) [inline lexer.go]: s/SetWindowFunc/SetSupportWindowFunc/?

Comment (ngaut) [reply lexer.go]: EnableWIndowFunc should be better.

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Comment by zz-jason** [comment]
please add this PR in https://github.com/pingcap/tidb/projects/18

**Pull request merged**